src/teehr/loading/nwm/retrospective_points.py

```diff
@@ -1,4 +1,29 @@
-"""A module for loading retrospective NWM point data."""
+"""A module for loading retrospective NWM point data (i.e., streamflow).
+
+The function ``nwm_retro_to_parquet()`` can be used to fetch and format three
+different versions of retrospective NWM data (v2.0, v2.1, and v3.0) for
+specified variables, locations, and date ranges.
+
+Each version of the NWM data is hosted on `AWS S3
+<https://registry.opendata.aws/nwm-archive/>`__ and is stored in Zarr format.
+Several options are included for fetching the data in chunks, including by
+week, month, or year, which can be specified using the ``chunk_by`` argument.
+
+Care must be taken when choosing a ``chunk_by`` value to minimize the amount
+of data transferred over the network.
+
+The Zarr stores for each version of the NWM data have the same internal
+chunking scheme: {"time": 672, "feature_id": 30000}, which defines the minimum
+amount of data that can be accessed at once (a single chunk). This means that
+if you specify ``chunk_by='week'``, the entire chunk
+(672 hours x 30000 locations) will be fetched for each week (128 hours)
+falling within that chunk, resulting in redundant data transfer.
```

Comment:
@samlamont Should we include something like, "it is currently not possible to avoid all redundant data transfer, but it should be minimized by selecting the largest chunk size that your computer's memory can handle" or something,